PSEUDO API 1.md

```diff
@@ -0,0 +1,272 @@
+
+### Todo
+- Need to expose the stage origin and bounds
+- configuration (e.g. change white balance on camera input, change options on map view)
+
+## XRPointCloud
+
+	interface XRPointCloud {
+		readonly attribute Float32Array points;
+	}
+
+## XRLightEstimate
+
+	interface XRLightEstimate {
+		readonly attribute double ambientIntensity;
+		readonly attribute double ambientColorTemperature;
+	}
+
+## XRAnchor
+
+	interface XRAnchor {
+		readonly attribute DOMString uid;
+		readonly attribute XRCoordinateSystem coordinateSystem;
+		readonly attribute double x;
+		readonly attribute double y;
+		readonly attribute double z;
+	}
+
+## XRPlaneAnchor
+
+	interface XRPlaneAnchor : XRAnchor {
+		readonly attribute Float32Array orientation; // quaternion?
+		readonly attribute double width;
+		readonly attribute double length;
+	}
+
+## XRManifold
+
+	interface XRManifold {
+		TBD
+	}
+
+### Todo
+
+- expose the manifold vertices and edges as well as its extent (FOV only, full sphere, etc)
+
+## XRPresentationFrame
+
+	interface XRPresentationFrame {
+		readonly attribute FrozenArray<XRView> views;
+
+		readonly attribute boolean hasPointCloud;
+		readonly attribute XRPointCloud pointCloud;
+
+		readonly attribute boolean hasManifold;
+		readonly attribute XRManifold manifold;
+
+		readonly attribute boolean hasLightEstimate;
+		readonly attribute XRLightEstimate lightEstimate;
+		
+		long addAnchor(XRAnchor anchor);
+		void removeAnchor(DOMString uid);
+		XRAnchor? getAnchor(DOMString uid);
+		<sequence <XRAnchor>> getAnchors();
+
+		XRDisplayPose? getDisplayPose(XRCoordinateSystem coordinateSystem);
+	};
+
+_The XRPresentationFrame differs from the VRPresentationFrame with the addition of the point cloud, manifold, light estimates, and anchor management._
+
+### Todo
+
+- access camera image buffer aor texture
+
+## XRView
+
+	interface XRView {
+		readonly attribute XREye eye;
+		readonly attribute Float32Array projectionMatrix;
+
+		XRViewport? getViewport(XRLayer layer);
+	};
+
+## XRViewport
+
+	interface XRViewport {
+		readonly attribute long x;
+		readonly attribute long y;
+		readonly attribute long width;
+		readonly attribute long height;
+	};
+
+
+## XRCartographicCoordinates
+
+	interface XRCartographicCoordinates {
+		attribute double latitude;
+		attribute double longitude;
+		attribute double positionAccuracy;
+		attribute double altitude;
+		attribute double altitudeAccuracy;
+		attribute Float32Array orientation; // quaternion from 0,0,0,1 EUS?
+	}
+
+The XRCartographicCoordinates are used in conjunction with the XRCoordinateSystem to represent a frame of reference that may optionally be positioned in relation to the nearest planet.
+
+## XRCoordinateSystem
+
+	interface XRCoordinateSystem {
+		readonly attribute XRCartographicCoordinates? mapLocation;
+
+		Float32Array? getTransformTo(XRCoordinateSystem other);
+	};
+
+## XRFrameOfReference
+
+	enum XRFrameOfReferenceType { "headModel", "eyeLevel", "stage", "spatial" };
```

Comment:
These types correspond to the attached, sitting, room scale, and spatial types used by Hololens.